beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -55,39 +55,145 @@ func (s *Server) ListAttestations(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	attestations = append(attestations, unaggAtts...)
+
 	isEmptyReq := rawSlot == "" && rawCommitteeIndex == ""
+	bothDefined := rawSlot != "" && rawCommitteeIndex != ""
+	var attsData json.RawMessage
+
 	if isEmptyReq {
 		allAtts := make([]*structs.Attestation, len(attestations))
 		for i, att := range attestations {
 			a, ok := att.(*eth.Attestation)
-			if ok {
-				allAtts[i] = structs.AttFromConsensus(a)
-			} else {
-				httputil.HandleError(w, fmt.Sprintf("unable to convert attestations of type %T", att), http.StatusInternalServerError)
+			if !ok {
+				httputil.HandleError(w, fmt.Sprintf("Unable to convert attestations of type %T", att), http.StatusInternalServerError)
 				return
 			}
+			allAtts[i] = structs.AttFromConsensus(a)
+		}
+		attsData, err = json.Marshal(allAtts)
+		if err != nil {
+			httputil.HandleError(w, "Could not marshal attestations: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+	} else {
+		filteredAtts := make([]*structs.Attestation, 0, len(attestations))
+		for _, att := range attestations {
+			committeeIndexMatch := rawCommitteeIndex != "" && att.GetData().CommitteeIndex == primitives.CommitteeIndex(committeeIndex)
+			slotMatch := rawSlot != "" && att.GetData().Slot == primitives.Slot(slot)
+			shouldAppend := (bothDefined && committeeIndexMatch && slotMatch) || (!bothDefined && (committeeIndexMatch || slotMatch))
+			if shouldAppend {
```

Comment:
I think this can be simplified like this:
```
committeeIndexMatxh := true
slotMatch := true
if rawCommitteeIndex != "" && att.GetData().CommitteeIndex != primitives.CommitteeIndex(committeeIndex) {
    committeeIndexMatxh = false
}
if rawSlot != "" && att.GetData().Slot != primitives.Slot(slot) {
    slotMatch = false
}

if committeeIndexMatch && slotMatch {
```